npeps/npep-126-egress-traffic-control.md

```diff
@@ -76,13 +76,196 @@ selected cluster workloads to k8s-apiservers for securing the server.
+    ports:
+      - portNumber:
+          protocol: TCP
+          port: 443
+      - portNumber:
+          protocol: TCP
+          port: 6443
+```
+
+### Implementing egress traffic control towards CIDRs
+
+This NPEP proposes to add a new type of `AdminNetworkPolicyEgressPeer` called `Networks`
+to be able to select destination CIDRs. This is provided to be able to select entities
+outside the cluster that cannot be selected using the other peer types.
+This peer type will not be supported in `AdminNetworkPolicyIngressPeer`.
+
+```
+// AdminNetworkPolicyEgressPeer defines a peer to allow traffic to.
+// Exactly one of the selector pointers must be set for a given peer. If a
+// consumer observes none of its fields are set, they must assume an unknown
+// option has been specified and fail closed.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+type AdminNetworkPolicyEgressPeer struct {
+    <snipped>
+	// Networks defines a way to select peers via CIDR blocks. This is
+	// intended for representing entities that live outside the cluster,
+	// which can't be selected by pods and namespaces peers, but note
+	// that cluster-internal traffic will be checked against the rule as
+	// well, so if you Allow or Deny traffic to `"0.0.0.0/0"`, that will allow
+	// or deny all IPv4 pod-to-pod traffic as well. If you don't want that,
+	// add a rule that Passes all pod traffic before the Networks rule.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	Networks []string `json:"networks,omitempty" validate:"omitempty,dive,cidr"`
+}
+```
+
+Note: It is recommended to use `networks` to select a set of CIDR range destinations
+that represent entities outside the cluster. If a user puts a podCIDR, nodeCIDR,
+serviceCIDR or other intra-cluster networks, it will work but it is better to use
+namespaces, pods, nodes peers to express such entities. Not all implementations can
+correctly define the boundary between "internal" and "external" destinations with respect
+to a Kubernetes cluster which is why this field is generic enough to select any CIDR
+destination.
+
+Example: Let's define ANP and BANP that refer to some CIDR networks:
+```
+apiVersion: policy.networking.k8s.io/v1alpha1
+kind: AdminNetworkPolicy
+metadata:
+  name: network-as-egress-peer
+spec:
+  priority: 70
+  subject:
+    namespaces: {}
+  egress:
+  - name: "deny-egress-to-external-dns-servers"
+    action: "Deny"
+    to:
+    - networks:
+      - 194.0.2.0/24
+      - 205.0.113.15/32
+      - 199.51.100.10/32
+    ports:
+      - portNumber:
+          protocol: UDP
+          port: 53
+  - name: "allow-all-egress-to-intranet"
+    action: "Allow"
+    to:
+    - networks:
+      - 192.0.2.0/24
+      - 203.0.113.0/24
+      - 198.51.100.0/24
+  - name: "allow-all-intra-cluster-traffic"
+    action: "Allow"
+    to:
+    - networks:
+      - POD_CIDR
```

Comment:
I'm assuming this is a place holder for actually CIDRs rather than some env vars?

Reply:
yes just a placeholder for the the actual podCIDR, thanks @Dyanngg for reviewing.